stack_test.go

```diff
@@ -0,0 +1,22 @@
+package errors
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCaller(t *testing.T) {
+	s := NewStackTrace(0)
+	assert.NotEmpty(t, s.StackTrace)
+
+	assert.Equal(t, fmt.Sprintf("%s", s), "[stack_test.go testing.go asm_amd64.s]")
```

Comment:
The odds are low, that's why it is even more important to do something about it, as it will be harder to debug in the future (or even make deployment impossible without having to modify the code).

I recommend one of the following easy solutions:

- When checking, only check for the first few files (stack_test.go and testing.go), ignoring the last ones (arch dependant).

- make the test portable by only running it on amd64 hosts by moving the test to a new test file called foo_amd64_test.go.

- use a type switch on ` sys.GOARCH`. Getting the correct results for all arch is complicated, as we don't have them, but a nice default with a remainder to add this arch to the test base will be enough.